src/snowcli/snow_connector.py

```diff
@@ -18,15 +19,25 @@
 
 
 def connect_to_snowflake(connection_name: Optional[str] = None, **overrides) -> SnowflakeConnection:  # type: ignore
-    connection_name = (
-        connection_name if connection_name is not None else get_default_connection()
+
+    context = (
+        snowcli.cli.common.snow_cli_global_context.snow_cli_global_context_manager.get_global_context_copy()
```

Comment:
IMO you shouldn't have to access global context here.

This method `connect_to_snowflake` is already invoked from the part of global context (`SnowCliGlobalContext.connection` -> `ConnectionDetails` -> `build_connection`).

IMO `--temporary-connection` flag should update `ConnectionDetails` like other connection flags, and then you can pass the value of this flag to `connect_to_snowflake` method.

Reply:
Changed